packages/sdk/react-client/src/client/ClientContext.tsx

```diff
@@ -78,55 +80,81 @@ export const ClientProvider = ({
   config: configProvider,
   services: createServices,
   client: clientProvider,
-  fallback = null,
+  fallback: Fallback = () => null,
   onInitialize
 }: ClientProviderProps) => {
-  const [client, setClient] = useState<Client | undefined>(
-    clientProvider instanceof Client ? clientProvider : undefined
-  );
+  const [client, setClient] = useState(clientProvider instanceof Client ? clientProvider : undefined);
+  const [status, setStatus] = useState<Status>();
   const [error, setError] = useState();
 
   if (error) {
     throw error;
   }
 
-  useAsyncEffect(async () => {
+  useEffect(() => {
+    if (!client) {
+      return;
+    }
+
+    const interval = setInterval(async () => {
```

Comment:
generally to be safe i'd move interval outside of effect and clear it each time effect is run. realize in this case client isn't likely to change.

Reply:
Not sure I understand, why outside the effect? Having it inside the effect with the clear on unmount ensures that the setInterval and clearInterval are paired